lib/models/BucketLoggingStatus.ts

```diff
@@ -0,0 +1,137 @@
+import { parseStringPromise } from 'xml2js';
+import errors, { ArsenalError } from '../errors';
+
+/**
+ * Format of xml request:
+<?xml version="1.0" encoding="UTF-8"?>
+<BucketLoggingStatus xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+   <LoggingEnabled>
+      <TargetBucket>string</TargetBucket>
+      <TargetGrants>
+         <Grant>
+            <Grantee>
+               <DisplayName>string</DisplayName>
+               <EmailAddress>string</EmailAddress>
+               <ID>string</ID>
+               <xsi:type>string</xsi:type>
+               <URI>string</URI>
+            </Grantee>
+            <Permission>string</Permission>
+         </Grant>
+      </TargetGrants>
+      <TargetObjectKeyFormat>
+         <PartitionedPrefix>
+            <PartitionDateSource>string</PartitionDateSource>
+         </PartitionedPrefix>
+         <SimplePrefix>
+         </SimplePrefix>
+      </TargetObjectKeyFormat>
+      <TargetPrefix>string</TargetPrefix>
+   </LoggingEnabled>
+</BucketLoggingStatus>
+*/
+
+export type LoggingEnabled = {
+    TargetBucket: string;
+    TargetPrefix: string;
+};
+
+export default class BucketLoggingStatus {
+    private _loggingEnabled?: LoggingEnabled;
+
+    constructor(loggingEnabled?: LoggingEnabled) {
+        this._loggingEnabled = loggingEnabled;
+    }
+
+    toXML(): string {
+        let loggingEnabledXML = "";
+        if (this._loggingEnabled) {
+            loggingEnabledXML = `<LoggingEnabled>
+    <TargetBucket>${this._loggingEnabled.TargetBucket}</TargetBucket>
+    <TargetPrefix>${this._loggingEnabled.TargetPrefix}</TargetPrefix>
+</LoggingEnabled>
+`;
+        }
+
+        return `<?xml version="1.0" encoding="UTF-8"?>
+<BucketLoggingStatus xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+    ${loggingEnabledXML}
+</BucketLoggingStatus>
+`;
+    }
+
+    static async fromXML(
+        data: string,
+    ): Promise<{ error?: { arsenalError: ArsenalError, details: any }; res?: BucketLoggingStatus; }> {
+        let parsed;
+        try {
+            parsed = await parseStringPromise(data);
+        } catch (err) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: err },
+                res: undefined,
```

Comment:
I assume we probably don't need to add an undefined `res` property (if we do, then maybe we need to revisit how we interpret the results to avoid being too javascript-centric).